### PR TITLE
Add Resource ome_application_csr and modify datasource ome_application_certificate sans schema

### PR DESCRIPTION
**Pull request by rounak-adhikary**
# Description
Add Resource ome_application_csr
Make subject_alternate_names a list of strings in DatSource ome_application_certificate

# ISSUE TYPE
Feature Pull Request

##### RESOURCE OR DATASOURCE NAME
 Resource ome_application_csr

##### OUTPUT

```
root@lglap049:~/terraform-provider-ome# go test -v -run="(TestDataSource_ReadCert|TestCsr)" terraform-provider-ome/ome -coverprofile=cov.out
=== RUN   TestDataSource_ReadCert
--- PASS: TestDataSource_ReadCert (2.79s)
=== RUN   TestCsr
--- PASS: TestCsr (8.08s)
PASS
coverage: 7.7% of statements
ok      terraform-provider-ome/ome      10.881s coverage: 7.7% of statements
root@lglap049:~/terraform-provider-ome#
```
image

# Checklist:

- [x] I have performed a self-review of my own code to ensure there are no formatting, vetting, linting, or security issues
- [x] I have verified that new and existing unit tests pass locally with my changes
- [x] I have not allowed coverage numbers to degenerate
- [x] I have maintained at least 90% code coverage
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have maintained backward compatibility

### CODE REVIEW

**Comment by terraform-providers-svc** [comment]
Can one of the admins verify this patch?

File: ome/resource_application_csr.go [hunk 1]
```diff
@@ -0,0 +1,217 @@
...
+	}
+	r.p = req.ProviderData.(*omeProvider)
+}
+
+// Metadata implements resource.Resource
+func (r resourceCsr) Metadata(ctx context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "application_csr"
+}
+
+// Template Deployment Resource schema
```
Comment (shenda1) [inline ome/resource_application_csr.go:57]: Update the comment accordingly

**Comment by shenda1** [comment]
LGTM

File: ome/resource_application_csr.go (hunk 1 above)
Comment (rounak-adhikary) [reply ome/resource_application_csr.go]: done

**Review by Krishnan-Priyanshu — APPROVED** [review]

**Review by shenda1 — APPROVED** [review]

**Pull request merged**